lunatrace/bsl/hasura/migrations/lunatrace/1652400983913_add-pr-comment-disable/up.sql

```diff
@@ -1,60 +1,19 @@
-ALTER TABLE public.settings ADD COLUMN pr_feedback_enabled boolean DEFAULT true NOT NULL;
-ALTER TABLE public.settings DROP COLUMN is_org_settings;
-ALTER TABLE public.projects ADD CONSTRAINT project_settings_fkey FOREIGN KEY (settings_id) REFERENCES public.settings;
-ALTER TABLE public.organizations ADD CONSTRAINT organization_settings_fkey FOREIGN KEY (settings_id) REFERENCES public.settings;
+ALTER TABLE public.settings ADD COLUMN pr_feedback_disabled boolean DEFAULT NULL;
 
---  get caught up on the missing settings rows
-DO $$
-    declare
-        project record;
-    BEGIN
-FOR project IN SELECT * FROM public.projects WHERE settings_id IS NULL
-LOOP
-    WITH setting AS (INSERT INTO public.settings DEFAULT VALUES RETURNING id)
-    UPDATE public.projects
-    SET settings_id=setting.id
-    FROM setting
-    WHERE public.projects.id = project.id;
-END LOOP;
-END; $$;
-
-DO $$
-    declare
-        organization record;
-    BEGIN
-        FOR organization IN SELECT * FROM public.organizations WHERE settings_id IS NULL
-            LOOP
-                WITH setting AS (INSERT INTO public.settings DEFAULT VALUES RETURNING id)
-                UPDATE public.organizations
-                SET settings_id=setting.id
-                FROM setting
-                WHERE public.organizations.id = organization.id;
-            END LOOP;
-    END; $$;
-
-
--- make sure any new projects or orgs have a settings table associated
-CREATE OR REPLACE FUNCTION always_create_settings()
-    RETURNS trigger AS
+CREATE OR REPLACE FUNCTION public.insert_default_settings()
+    RETURNS uuid AS
 $BODY$
+DECLARE new_id uuid;
 BEGIN
-    if (NEW.settings_id IS NULL) then
-        INSERT INTO public.settings DEFAULT VALUES RETURNING id INTO NEW.settings_id;
-    end if;
-    RETURN NEW;
+    INSERT INTO public.settings DEFAULT VALUES RETURNING id INTO new_id;
```

Comment:
This may not solve the issue with nullability. If we're trying to implement inheritance, inserting the default values for every project will override org-wide settings in all cases.

Reply:
What about if the columns in the settings table are nullable? Then there can be a fallback to the organization settings.

Regardless, given that we don't actually have inheritance implemented yet, I vote that we punt this and just pick the easy solution that is future proof. (Even if that means only _new_ projects get the inheritance because the current values are non-nullable, that's fine.)

Reply:
> that is precicely how it is implemented as you see here

Reply:
>  If we're trying to implement inheritance, inserting the default values for every project will override org-wide settings in all cases.
The defaults ARE null values so I think that fixes it